Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RemoteAssetsDownloader.java

```diff
@@ -111,4 +139,100 @@ public String getService(String service) {
             return "in.yatri.provider";
         }
     }
-    }
\ No newline at end of file
+
+    public String appendSdkNameAndVersion(String file) {
+        final int dotPosition = file.lastIndexOf('.');
+        int versionId = getApplicationContext().getResources().getIdentifier("godel_version", "string", getApplicationContext().getPackageName());
+        String version = getApplicationContext().getString(versionId);
+        int nameId = getApplicationContext().getResources().getIdentifier("godel_app_name", "string", getApplicationContext().getPackageName());
+        String name = getApplicationContext().getString(nameId);
+
+        if (dotPosition > 0 && dotPosition < file.length() - 1) {
+            String fileName = file.substring(0, dotPosition);
+            String extension = file.substring(dotPosition);
+
+            return fileName + "_" + name + "_" + version + extension;
+        } else {
+            return file + "_" + name + "_" + version;
+        }
+    }
+
+    public int updateBundle(Intent intent){
+        try {
+            JSONArray urls = new JSONArray(intent.getStringExtra("urls"));
+            for (int i = 0; i < urls.length(); i++) {
+                String url = urls.getJSONObject(i).has("url") ? urls.getJSONObject(i).getString("url") : null;
+                String fileName = "";
+                int index = url.lastIndexOf("/");
+                if (index != -1) {
+                    fileName = url.substring(index + 1);
+                    fileName = appendSdkNameAndVersion(fileName);
+                }
+                String path = urls.getJSONObject(i).has("path") ? urls.getJSONObject(i).getString("path") : null;
+                String finalPath = "";
+                if (!path.isEmpty()) {
+                    char lastChar = path.charAt(path.length() - 1);
+                    if(lastChar == '/'){
+                        finalPath = path+fileName;
+                    }else{
+                        finalPath = path+'/'+fileName;
+                    }
+                } else {
+                    finalPath = fileName;
+                }
+                String description = urls.getJSONObject(i).has("description") ? urls.getJSONObject(i).getString("description") : "";
+                String title = urls.getJSONObject(i).has("title") ? urls.getJSONObject(i).getString("title") : "";
+                String image = urls.getJSONObject(i).has("image") ? urls.getJSONObject(i).getString("image") : "";
+                JSONObject payloadData = new JSONObject();
+                payloadData.put("description",description);
```

Comment:
one Notification should show only one pop up it will be better to send title ,description and image outside the urls.